Expose a few functions from fpdf_text.h for getting text on a page.

billmoyers opened the pull request:
Hi, this exposes some of the FPDFText_* functions which are useful for getting text. Thanks.

billmoyers committed:
Expose a few functions from fpdf_text.h for getting text on a page.

billmoyers committed:
Expose a few functions from fpdf_text.h for getting text on a page.

billmoyers committed:
Expose a few functions from fpdf_text.h for getting text on a page.

Commit:
fixed potential NPE

barteksc commented:
What is the purpose of `nativeBitmapFillRect()` function? You can fill bitmap with color using Android API

barteksc committed:
Merge pull request fixing potential NPE when getting links
Update README and CHANGELOG
Update version

billmoyers commented:
My impression was that FPDFBitmap_FillRect would fill in between the text and the background layer which I believe would be difficult to do after the bitmap has been rendered by FPDF.  Maybe I missed something?

barteksc commented:
There are no layers, this just a bitmap. FillRect will fill given region with given color

billmoyers commented:
I just realized that I didn't mean to include the FillRect in the pull request anyway because I hadn't really tested it.  I removed the FillRect commit from this PR.  The other commit was tested.  How does that one look?

barteksc commented:
Thank you for your work, I will review commit source and add it to next release

johngray1965 commented on src/main/java/com/shockwave/pdfium/PdfiumCore.java:
```diff
@@ -440,4 +466,201 @@ public RectF mapRectToDevice(PdfDocument doc, int pageIndex, int startX, int sta
+        return 0;
+    }
+
+    public class Rect {
```
Other places seem to using android's RectF?

johngray1965 reviewed (commented):
Could we also expose the device to page mapping functions?

billmoyers committed:
Remove unnecessary new Rect class, just use RectF from Android.

billmoyers committed:
Remove unnecessary new Rect class, just use RectF from Android.

billmoyers commented:
@johngray1965 is right, we don't need the extra class. I removed it for RectF and pushed the commit.

barteksc committed:
Use pdfium built with c++_shared
Update plugins
Update compile SDK and support libs to 26
Set minimum SDK to 14
Update version and README

barteksc committed:
Update CHANGELOG

envolva commented:
Is this feature still in the Works? We would like to use this with Pdfium. Thanks.

johngray1965 commented:
I've been using this branch, I'd love to see it get merged.

Commit:
Add support to close the text pages.
Fixed a few crashes
Fixed an issue with openTextPage leaking the page.

Commit:
Add support to close the text pages.
Fixed a few crashes
Fixed an issue with openTextPage leaking the page.

Commit:
Merge pull request #2 from johngray1965/master

Add support to close the text pages.

Commit:
When we open up search, give it focus.
Fixed an issue with Google voices that are Oreo or later, but still don't do work events.

Commit:
Fixed an issue in the pdf password handling.

oceanwaves90 commented:
1. Is it needed to clear mNativeTextPagesPtr and call nativeCloseTextPages for each of them when disposing?

2. And I wonder why openTextPage calls openPage also. Are textPages different from usual pages?(why does it load usual pages too?)

3. What does 'length' mean in textPageGetBoundedText? How long do you recommend for me?

Commit:
Add the proper icons for just about everything.
Tied the four settings buttons on the camera screen to the settings
Implemented email support, twitter, facebook, the reader app, write a review (which only takes the user to play store page for the app, that's all we can do there).

Commit:
Upgraded crux, which fixed a user issue.

Commit:
When the user pauses make sure playOnFocus gets set to false.

Commit:
Merge branch 'master' of https://github.com/billmoyers/PdfiumAndroid

Commit:
Merge branch 'master' of https://github.com/billmoyers/PdfiumAndroid

Commit:
Add basic wrapper for FPDF_SaveAsCopy.

Commit:
Merge branch 'master' of https://github.com/billmoyers/PdfiumAndroid

Commit:
Expose getFontSize and getPageCropBox.

Commit:
Change getCropBox to use a double arrray in the native code.

Commit:
Updated the version

Commit:
Update versionName to "3.4.11"

Commit:
Update gradle plugin
Formatting

Commit:
Set scanner's versionName to "1.0.26"
Fixed dependencies issues.

Commit:
Fixed Bookshare obfuscation issue.

Commit:
Update gradle build tools to 4.1.1

Commit:
Get Voices from Legere's docConversion/AWS infrastructure.

Commit:
Pdf Margins mostly working now.

Commit:
Fix the fact that Sharon is now called Sharona.

Commit:
Moved the dependency injection to hilt.
Fixed some issues in the tablet view for the Pronunciations.
Fixed issues with the pdf view initial zoom (fixed fuzziness)

Commit:
Update to the new billing library for purchasing voices.
Fix an NPE in Daisy audio import.
